blackbird_python/blackbird/program.py

```diff
@@ -363,8 +383,12 @@ def serialize(self):
                         array_insert += len(bb_array)
 
                     elif isinstance(v, str):
-                        # argument is a string type
-                        args.append('"{}"'.format(v))
+                        # argument is a string type; if a tdm loop parameter ("BS", "M" or "R"),
+                        # then simply add it as is
+                        if self.programtype["name"] == "tdm" and ["BS", "M", "R"]:
```

Comment:
(Minor suggestion): maybe worth making `["BS", "M", "R"]` a tuple/frozenset (immutable collections can be faster) and have it on the top level of the file, such that it only needs to be changed at one place.